src/main/java/org/spongepowered/common/mixin/core/world/MixinDimensionType.java

```diff
@@ -0,0 +1,107 @@
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.core.world;
+
+import net.minecraft.world.DimensionType;
+import net.minecraft.world.WorldProvider;
+import org.spongepowered.api.service.context.Context;
+import org.spongepowered.api.world.Dimension;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Implements;
+import org.spongepowered.asm.mixin.Interface;
+import org.spongepowered.asm.mixin.Intrinsic;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.common.SpongeImpl;
+import org.spongepowered.common.config.SpongeConfig;
+import org.spongepowered.common.interfaces.world.IMixinDimensionType;
+import org.spongepowered.common.world.DimensionManager;
+
+import java.util.Optional;
+
+@Mixin(DimensionType.class)
+@Implements(value = @Interface(iface = org.spongepowered.api.world.DimensionType.class, prefix = "dimensionType$"))
+public abstract class MixinDimensionType implements IMixinDimensionType {
+
+    @Shadow @Final private Class <? extends WorldProvider> clazz;
+    @Shadow public abstract String getName();
+
+    private String sanitizedId;
+    private SpongeConfig<SpongeConfig.DimensionConfig> config;
+    private volatile Context context;
+
+    @Inject(method = "<init>", at = @At("RETURN"))
+    public void onConstruct(String enumName, int ordinal, int idIn, String nameIn, String suffixIn, Class <? extends WorldProvider > clazzIn,
+            CallbackInfo ci) {
+        this.sanitizedId = this.getName().toLowerCase().replaceAll(" ", "_").replace("[^A-Za-z0-9_]", "");
```

Comment:
Single char replacements should use `.replace(' ', '_')`, also I don't think the regex replace here is working because `replace` does only literal replacements, `replaceAll` is the method which supports regular expressions